lib/spack/spack/util/windows_registry.py

```diff
@@ -182,7 +181,7 @@ def _load_key(self):
                 winreg.OpenKeyEx(self.root.hkey, self.key, access=winreg.KEY_READ),
             )
         except FileNotFoundError as e:
-            if e.winerror == 2:
+            if sys.platform == "win32" and e.winerror == 2:
```

Comment:
Note this extra check is unnecessary because this object cannot be instantiated on non-windows systems

Reply:
It’s another unfortunate mypy thing. The winerror attribute doesn’t exist on non-windows systems but mypy sees it statically, so without this you get an error at typecheck time.